stdlib/public/core/UnicodeHelpers.swift

```diff
@@ -395,17 +395,31 @@ extension _StringGuts {
         startingAt: String.Index(_encodedOffset: start)
       ).0))
     }
+    let range = start..<end
+    let from = self._object.cocoaObject
+
+    let capacity = 16 * 4
```

Comment:
What is the relationship between this capacity and the actual fixed array size? Why this capacity? Seems like 16 code units would be more than enough for anything non-pathological. What use cases were you seeing?

Reply:
Do you mean this will be enough?

```
// 16 code units would be more than enough for anything non-pathological
_internalInvariant(count <= 16 * MemoryLayout<UInt64>.stride)
var cus = _FixedArray16<UInt64>(allZeros: ())
return cus.withUnsafeMutableBufferPointer { buffer in
  let baseAddress = buffer.baseAddress._unsafelyUnwrappedUnchecked
  let into = UnsafeMutableRawPointer(baseAddress)
    .bindMemory(to: UInt16.self, capacity: 16 * MemoryLayout<UInt64>.stride)
  _cocoaStringCopyCharacters(from: from, range: range, into: into)
  let intoBuffer = UnsafeBufferPointer(start: UnsafePointer<UInt16>(into), count: range.count)
  return Character(String._uncheckedFromUTF16(intoBuffer))
}
```